state/factory/statedb.go

```diff
@@ -42,6 +42,7 @@ type stateDB struct {
 	dao                db.KVStore // the underlying DB for account/contract storage
 	timerFactory       *prometheustimer.TimerFactory
 	workingsets        *lru.Cache // lru cache for workingsets
+	protocolView       protocol.Dock
```

Comment:
view should be loaded on start for both statedb and factory

Reply:
not sure if i understand correctly, i think view should be loaded by the protocol upon receiving the workingset? b/c the value stored is a generic interface{}, it's up to each protocol to load from ws and convert it

also if the block does not contain staking actions there's actually no need for statedb/factory to load the view, so it should be loaded by its consumer, which is protocol. that is my understanding

Reply:
factory always stays in memory.  if the view is loaded in the fly, it may take a long time.
Two cases for a protocol's view:
1. It is used in a block: we need to load it any way
2. it is not used in a block: the view won't be updated, it doesn't hurt to keep it as it was in the factory.

Reply:
yes, factory always in-mem, i think the "load" we discuss here refers to constructing the cand center from candidates in the factory, and keep it in the factory's Dock, right?

2 cases for our staking protocol:
1. bootstrap cand list != nil, then CreateGenesisState() will do the "load", and store the cand center into factory's Dock, Dock["staking"] = center
2. bootstrap cand list == nil, "load" will be done by staking protocol on the 1st block that contains staking action

and after that, read view (in-the-fly) is just an in-mem map access: center = factory.Dock["staking"], won't take long time

since "load" is only a one-time action, i think can let each protocol handle its own. factory just provide the Dock, each protocol will only "load" once and later it's a quick memory access to load the view

if we want to do "load" in factory, i feel there are 2 concerns:
1. factory needs to call staking protocol's method to construct the cand center, in our case createCandCenter(), but factory shouldn't understand these details
2. if we have more protocols in the future, factory needs to know all the protocols and "load" all of them in Start(), again factory no need to know all these protocols, it just provide a Dock for protocols to read/write a generic interface{}, that's it